src/julia_to_gap.jl

```diff
@@ -215,26 +223,32 @@ end
 ## and if `obj` contains Julia subobjects;
 ## in this case, `obj` is a GAP list or record.
 ## An example of such an `obj` is `GAP.julia_to_gap([[1]])`.
-function julia_to_gap(
+function julia_to_gap_internal(
     obj::GapObj,
-    recursion_dict::IdDict{Any,Any} = IdDict();
-    recursive::Bool = false,
+    recursion_dict::GapCacheDict,
+    recursive::Bool,
 )
     if ! recursive
         ret_val = obj
     elseif Wrappers.IsList(obj)
         len = length(obj)
         ret_val = NewPlist(len)
+        if recursion_dict === nothing
+          recursion_dict = RecDict()
+        end
         recursion_dict[obj] = ret_val
         for i = 1:len
-             ret_val[i] = julia_to_gap(obj[i], recursion_dict; recursive)
+             ret_val[i] = julia_to_gap_internal(obj[i], recursion_dict, recursive)
```

Comment:
Another case with missing `get!(recursion_dict::RecDict, x)` etc.?

Reply:
Yes, good spot!
I will add a test for this situation (a GAP list containing identical Julia objects which themselves do *not* support/need recursion).